### PR TITLE
.github: create clusters with taints

### PR DESCRIPTION
**Pull request by aanm**
As recommended to users, cilium-cli should also set the clusters with
taints. This guarantees that Cilium is properly configured and
application pods can start successfully.

Signed-off-by: André Martins <andre@cilium.io>

### CODE REVIEW

**Review by joestringer — APPROVED** [review]
Looks fine for @cilium/github-sec . I'll let folks with more understanding of Azure comment on those bits.

File: .github/workflows/eks.yaml [hunk 1]
```diff
@@ -57,10 +57,34 @@ jobs:
 
       - name: Create EKS cluster without nodegroup
```
Comment (nbusseneau) [inline .github/workflows/eks.yaml:58]: Likewise.

File: .github/workflows/aks.yaml [hunk 2]
```diff
@@ -68,21 +68,43 @@ jobs:
...
+            --name nodepool2 \
+            --cluster-name ${{ env.name }} \
+            --resource-group ${{ env.name }} \
             --node-count 2 \
             --node-vm-size Standard_B2s \
             --node-osdisk-size 30 \
-            --load-balancer-sku basic \
-            --generate-ssh-keys
+            --mode system \
+            --node-taints node.cilium.io/agent-not-ready=true:NoSchedule
```
Comment (nbusseneau) [inline .github/workflows/aks.yaml:107]:
```suggestion
            --priority Spot \
            --eviction-policy Delete \
            --spot-max-price -1 \
            --mode system \
            --node-taints node.cilium.io/agent-not-ready=true:NoSchedule
```
Can we try adding these in? Now that we are using the `standard` load balancer and an additional nodepool anyway, let's try to use spot instances.

File: defaults/defaults.go [hunk 3]
```diff
@@ -76,7 +76,7 @@ const (
 	ConnectivityCheckJSONMockImage   = "quay.io/cilium/json-mock:v1.3.0@sha256:2729064827fa9dbfface8d3df424feb6c792a0ba07117b844349635c93c06d2b"
 
 	ConfigMapName = "cilium-config"
-	Version       = "v1.10.2"
+	Version       = "v1.10.3"
```
Comment (nebril) [inline defaults/defaults.go:79]: nitpick - this change should probably be in separate commit in case we need to revert it?

**Review by nebril — APPROVED** [review]

File: defaults/defaults.go (hunk 3 above)
Comment (joestringer) [reply defaults/defaults.go:79]: I think this has to go together, otherwise you may end up with clusters with taints that will never be removed.

**Review by twpayne — APPROVED** [review]

File: defaults/defaults.go (hunk 3 above)
Comment (aanm) [reply defaults/defaults.go:79]: Correct

File: .github/workflows/aks.yaml (hunk 2 above)
Comment (aanm) [reply .github/workflows/aks.yaml:112]: System node pool can not use spot VM

**Review by tklauser — APPROVED** [review]
